tools/checklocks/analysis.go

```diff
@@ -499,11 +499,7 @@ func (pc *passContext) checkInstruction(inst ssa.Instruction, ls *lockState) (*s
 
 // checkBasicBlock traverses the control flow graph starting at a set of given
 // block and checks each instruction for allowed operations.
-func (pc *passContext) checkBasicBlock(fn *ssa.Function, block *ssa.BasicBlock, lff *lockFunctionFacts, parent *lockState, seen map[*ssa.BasicBlock]*lockState) *lockState {
-	if oldLS, ok := seen[block]; ok && oldLS.isCompatible(parent) {
-		return nil
-	}
-
+func (pc *passContext) checkBasicBlock(fn *ssa.Function, block *ssa.BasicBlock, lff *lockFunctionFacts, parent *lockState) *lockState {
```

Comment:
The return value of this function is nil when the block does not return, otherwise it returns the return lock state of the block. I've added some comments to the function to clarify this.